terraform/main.tf

```diff
@@ -79,70 +79,149 @@ locals {
   additional_disks = concat(local.fs_disks, local.asm_disks)
 
   project_id = var.project_id
+
+  is_multi_instance = (
+    var.zone1 != "" && var.zone2 != "" && var.subnetwork1 != "" && var.subnetwork2 != ""
+  )
+
+  instances = local.is_multi_instance ? {
+    node1 = {
+      name       = "${var.instance_name}-1"
+      zone       = var.zone1
+      subnetwork = var.subnetwork1
+    }
+    node2 = {
+      name       = "${var.instance_name}-2"
+      zone       = var.zone2
+      subnetwork = var.subnetwork2
+    }
+  } : {
+    default = {
+      name       = var.instance_name
+      zone       = var.zone1
+      subnetwork = var.subnetwork1
+    }
+  }
 }
 
-module "instance_template" {
-  source  = "terraform-google-modules/vm/google//modules/instance_template"
-  version = "~> 13.0"
-
-  name_prefix        = format("%s-template", var.instance_name)
-  region             = var.region
-  project_id         = local.project_id
-  network            = coalesce(var.network, var.subnetwork)
-  subnetwork         = coalesce(var.subnetwork, var.network)
-  subnetwork_project = local.project_id
-  service_account = {
-    email  = var.vm_service_account
-    scopes = ["https://www.googleapis.com/auth/cloud-platform"]
+
+data "google_compute_image" "os_image" {
+  family  = var.source_image_family
+  project = var.source_image_project
+}
+
+resource "time_static" "template_suffix" {}
+
+locals {
+  template_suffix = formatdate("YYYYMMDDhhmmss", time_static.template_suffix.rfc3339)
+}
+
+resource "google_compute_instance_template" "default" {
+  name         = "${var.instance_name}-${local.template_suffix}"
+  project      = var.project_id
+  machine_type = var.machine_type
+
+  network_interface {
+    # gets overridden during instance creation
+    network = "default"
+  }
+  disk {
+    boot = true
+    auto_delete = true
+    source_image = data.google_compute_image.os_image.self_link
+    disk_type = var.boot_disk_type
+    disk_size_gb  = var.boot_disk_size_gb
   }
 
-  machine_type         = var.machine_type
-  source_image_family  = var.source_image_family
-  source_image_project = var.source_image_project
-  disk_size_gb         = var.boot_disk_size_gb
-  disk_type            = var.boot_disk_type
-  auto_delete          = true
+  dynamic "disk" {
+    for_each = local.additional_disks
+    content {
+      auto_delete       = lookup(disk.value, "auto_delete", null)
+      boot              = lookup(disk.value, "boot", null)
```

Comment:
If we use `boot_disk_type` and `boot_disk_size_gb` to define the boot disk parameters, perhaps we should remote the option to make an additional disk a boot disk.